{"version":3,"file":"ArgumentOutOfRangeError.js","sourceRoot":"","sources":["../../src/util/ArgumentOutOfRangeError.ts"],"names":[],"mappings":";;;;;;;;;;;;AAAA;;;;;;;;;GASG;AACH;IAA6C,2CAAK;IAChD;QAAA,iBAKC;QAJC,IAAM,GAAG,WAAQ,kBAAM,uBAAuB,CAAC,QAAA,CAAC;QACzC,KAAK,CAAC,IAAI,GAAG,GAAG,CAAC,IAAI,GAAG,yBAAyB,CAAC;QAClD,KAAK,CAAC,KAAK,GAAG,GAAG,CAAC,KAAK,CAAC;QACxB,KAAK,CAAC,OAAO,GAAG,GAAG,CAAC,OAAO,CAAC;;IACrC,CAAC;IACH,8BAAC;AAAD,CAAC,AAPD,CAA6C,KAAK,GAOjD;AAPY,0DAAuB","sourcesContent":["/**\n * An error thrown when an element was queried at a certain index of an\n * Observable, but no such index or position exists in that sequence.\n *\n * @see {@link elementAt}\n * @see {@link take}\n * @see {@link takeLast}\n *\n * @class ArgumentOutOfRangeError\n */\nexport class ArgumentOutOfRangeError extends Error {\n  constructor() {\n    const err: any = super('argument out of range');\n    (<any> this).name = err.name = 'ArgumentOutOfRangeError';\n    (<any> this).stack = err.stack;\n    (<any> this).message = err.message;\n  }\n}\n"]}